py3status/modules/imap.py

```diff
@@ -325,19 +360,58 @@ def _get_mail_count(self):
                         auth_string = "user={}\1auth=Bearer {}\1\1".format(
                             self.user, self.creds.token
                         )
-                        self.connection.authenticate("XOAUTH2", lambda x: auth_string)
+                        self.connection.authenticate(
+                            "XOAUTH2", lambda x: auth_string
+                        )
                     else:
                         # Login with user and password
                         self.connection.login(self.user, self.password)
 
                 tmp_mail_count = 0
-                directories = self.mailbox.split(",")
+                if self.mailbox:
+                    directories = self.mailbox.split(",")
+                else:
+                    try:
+                        directories
+                    except:
+                        directories = []
+                        filters = "|".join(self.filter)
```

Comment:
why not just pass a single regex in filter?

Reply:
that would certainly be possible but i think that 
```
filter = ['pattern']
filter += ['pattern2']
...
```
is a lot more readable than 
```
filter = 'pattern|pattern2'
```